internal/controllers/gateway/gateway_utils.go

```diff
@@ -351,6 +352,13 @@ func getListenerStatus(
 			})
 		}
 
+		// consistent sort statuses to allow equality comparisons
+		sort.Slice(status.Conditions, func(i, j int) bool {
+			a := status.Conditions[i]
+			b := status.Conditions[i]
```

Comment:
should be  `b := status.Conditions[j]` here

Reply:
:facepalm: